contrib/golang/filters/http/source/config.cc

```diff
@@ -33,7 +33,10 @@ Http::FilterFactoryCb GolangFilterConfig::createFilterFactoryFromProtoTyped(
       proto_config, dso_lib, fmt::format("{}golang.", stats_prefix), context);
   config->newGoPluginConfig();
   return [config, dso_lib](Http::FilterChainFactoryCallbacks& callbacks) {
-    auto filter = std::make_shared<Filter>(config, dso_lib);
+    const std::string worker_name = callbacks.dispatcher().name();
+    const uint32_t worker_id =
+        std::stoi(worker_name.substr(worker_name.find_first_of('_') + 1, worker_name.size()));
```

Comment:
can we have some sanity-checking on the return from find_first_of? At least an assert? I think this reliance the worker name syntax could crash the process. What's the 'main' dispatcher called?

Better yet, do a refactor to provide Dispatcher::id() and handle the case where it's the main thread.

also, the size passed into substr is supposed to be the length of the desired string, not the end position in the source string. You can just omit that param though.

Reply:
you want the system to crash if the worker thread name syntax changes?  Actually that's fine but use ENVOY_BUG.

I would have similar comments on the code you copied from but I didn't review that code :)